expression/integration_test.go

```diff
@@ -4930,7 +4930,55 @@ func (s *testIntegrationSuite) TestIssue10181(c *C) {
 
 func (s *testIntegrationSuite) TestExprPushdownBlacklist(c *C) {
 	tk := testkit.NewTestKit(c, s.store)
-	tk.MustQuery(`select * from mysql.expr_pushdown_blacklist`).Check(testkit.Rows())
+	tk.MustQuery(`select * from mysql.expr_pushdown_blacklist`).Check(testkit.Rows(
+		"date_add tiflash DST(daylight saving time) does not take effect in TiFlash date_add",
+		"cast tiflash Behavior of some corner cases(overflow, truncate etc) is different in TiFlash and TiDB"))
+
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t")
+	tk.MustExec("create table t(a int , b date)")
+
+	// Create virtual tiflash replica info.
+	dom := domain.GetDomain(tk.Se)
+	is := dom.InfoSchema()
+	db, exists := is.SchemaByName(model.NewCIStr("test"))
+	c.Assert(exists, IsTrue)
+	for _, tblInfo := range db.Tables {
+		if tblInfo.Name.L == "t" {
+			tblInfo.TiFlashReplica = &model.TiFlashReplicaInfo{
+				Count:     1,
+				Available: true,
+			}
+		}
+	}
+
+	tk.MustExec("insert into mysql.expr_pushdown_blacklist " +
+		"values('<', 'tikv,tiflash,tidb', 'for test'),('date_format', 'tikv', 'for test')")
+	tk.MustExec("admin reload expr_pushdown_blacklist")
+
+	tk.MustExec("set @@session.tidb_isolation_read_engines = 'tiflash'")
+
+	// < not pushed, cast only pushed to TiKV, date_format only pushed to TiFlash,
+	// > pushed to both TiKV and TiFlash
+	tk.MustQuery("explain select * from test.t where b > date'1988-01-01' and b < date'1994-01-01' " +
+		"and cast(a as decimal(10,2)) > 10.10 and date_format(b,'%m') = '11'").Check(testkit.Rows(
+		"Selection_5 2133.33 root  lt(test.t.b, 1994-01-01)",
+		"└─Selection_9 2666.67 root  gt(cast(test.t.a), 10.10)",
+		"  └─TableReader_8 2666.67 root  data:Selection_7",
+		"    └─Selection_7 2666.67 cop[tiflash]  eq(date_format(test.t.b, \"%m\"), \"11\"), gt(test.t.b, 1988-01-01)",
+		"      └─TableFullScan_6 10000.00 cop[tiflash] table:t keep order:false, stats:pseudo"))
```

Comment:
We'd better not to compare the `explain` result directly. That is not friendly enough for other developers. You can get the `testkit.rows` and compare the expression info in the second-to-last row.